python/tk_framework_desktopserver/shotgun/scripts/cache_commands.py

```diff
@@ -0,0 +1,125 @@
+# Copyright (c) 2017 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import sys
+import cPickle
+
+def cache(cache_file, data, base_configuration, hash_data, engine_name):
+    import sqlite3
+    import sgtk
+
+    entity = dict(type=data["entity_type"], id=data["entity_id"])
+    project_entity = dict(id=data["project_id"], type="Project")
+
+    # Setup the bootstrap manager.
+    toolkit_mgr = sgtk.bootstrap.ToolkitManager()
+    toolkit_mgr.plugin_id = "basic.shotgun.cache"
+    toolkit_mgr.base_configuration = base_configuration
+
+    pcs = toolkit_mgr.get_pipeline_configurations(
+        project=project_entity,
+    ) or [dict(id=None)]
+
+    for pc in pcs:
```

Comment:
This method currently returns an array of dictionaries. It would still return the same thing, only now there would be two extra fields. I think the polish ticket should be about adding those two fields and that's it.